fs/operations/resume_test.go

```diff
@@ -0,0 +1,165 @@
+package operations
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fstest"
+	"github.com/rclone/rclone/fstest/mockobject"
+	"github.com/rclone/rclone/lib/atexit"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type interruptReader struct{}
+
+// Read sends an OS specific interrupt signal, waits for the signal to be
+// received, and then simulates read being cancelled
+func (r *interruptReader) Read(b []byte) (n int, err error) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	err = sendInterrupt()
+	if err != nil {
+		return 0, err
+	}
+	go func() {
+		for {
+			if atexit.Signalled() {
+				wg.Done()
+				return
+			}
+		}
```

Comment:
`Signalled` returns immediately so this loop is in fact a short-living 100% cpu hog, a spin-poll. I suggest to add a small delay like `time.Sleep(10*time.Millisecond)` after the `if`.

Reply:
@mcalman
`Signalled` returns immediately so this loop is in fact a short-living spin-poll, 100% cpu hog.
Generally this means that you'd better revise your solution idea. I am leaving that decision to @ncw and yourself.
As a _quick fix_ I would add a small delay: (?)
```
go func() {
		for {
			if atexit.Signalled() {
				wg.Done()
				return
			}
			time.Sleep(10 * time.Millisecond)
		}
} 
```